src/deploy/AllocatorInit.sol

```diff
@@ -0,0 +1,203 @@
+}
+
+interface DepositorUniV3Like {
+    struct LiquidityParams {
+        address gem0;
+        address gem1;
+        uint24  fee;
+        int24   tickLower;
+        int24   tickUpper;
+        uint128 liquidity;
+        uint256 amt0Desired;
+        uint256 amt1Desired;
+        uint256 amt0Min;
+        uint256 amt1Min;
+    }
+
+    struct CollectParams {
+        address gem0;
+        address gem1;
+        uint24  fee;
+        int24   tickLower;
+        int24   tickUpper;
+    }
+
+    function deposit(LiquidityParams memory) external returns (uint128, uint256, uint256);
+    function withdraw(LiquidityParams memory, bool) external returns (uint256, uint256);
+    function collect(CollectParams memory) external returns (uint256, uint256);
+}
+
+interface KissLike {
+    function kiss(address) external;
+}
+
+struct AllocatorConfig {
+    uint256 debtCeiling;
+    address allocatorProxy;
+    uint8 facilitatorRole;
+    uint8 automationRole;
+    address facilitator;
+    address keeper;
+    address[] swapGems;
+    address[] depositGems;
+    bytes32 vaultClKey;
+    bytes32 bufferClKey;
+}
+
+library AllocatorInit {
+    uint256 constant WAD = 10 ** 18;
+    uint256 constant RAY = 10 ** 27;
+    uint256 constant RAD = 10 ** 45;
+
+    function switchOwner(address base, address currentOwner, address newOwner) internal {
+        if (currentOwner == newOwner) return;
+        require(WardsLike(base).wards(currentOwner) == 1, "AllocatorInit/current-owner-not-authed");
+        WardsLike(base).rely(newOwner);
+        WardsLike(base).deny(currentOwner);
+    }
+
+    // TODO: add to ilk registry?
+    // TODO: sanity checks
+
+    function initShared(
+        DssInstance memory dss,
+        AllocatorSharedInstance memory sharedInstance
+    ) internal {
+        dss.chainlog.setAddress("ALLOCATOR_ORACLE",   sharedInstance.oracle);
+        dss.chainlog.setAddress("ALLOCATOR_ROLES",    sharedInstance.roles);
+        dss.chainlog.setAddress("ALLOCATOR_REGISTRY", sharedInstance.registry);
+    }
+
+    function initAllocator(
+        DssInstance memory dss,
+        AllocatorSharedInstance memory sharedInstance,
+        AllocatorNetworkInstance memory networkInstance,
+        AllocatorConfig memory cfg
+    ) internal {
+        bytes32 ilk = VaultLike(networkInstance.vault).ilk();
+
+        // Onboard the ilk
+        dss.vat.init(ilk);
+        dss.jug.init(ilk);
+
+        require(cfg.debtCeiling < WAD, "AllocatorInit/incorrect-ilk-line-precision");
+        dss.vat.file(ilk, "line", cfg.debtCeiling * RAD);
+        dss.vat.file("Line", dss.vat.Line() + cfg.debtCeiling * RAD);
+
+        dss.spotter.file(ilk, "pip", sharedInstance.oracle);
+        dss.spotter.file(ilk, "mat", RAY);
+        dss.spotter.poke(ilk);
+
+        // Configure the Registry and Roles contracts
+        RolesLike(sharedInstance.roles).setIlkAdmin(ilk, cfg.allocatorProxy);
```

Comment:
As discussed, we first need to set it to the pause proxy and after all the roles and actions added set to the allocator proxy.